apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/backend/ir/BackendIrBuilder.kt

```diff
@@ -52,6 +54,42 @@ internal class BackendIrBuilder constructor(
     )
   }
 
+  /**
+   * a GroupedField is a list of fields with the same responseName and arguments but possibly different selectionSets and directives
+   */
+  private data class GroupedField(val fields: List<GQLField>)
```

Comment:
Why not just merger all fields into one field?

Reply:
A `GQLField` cannot represent exactly a list of field because:
```
{
  name @include(if: $condition1)
  name @include(if: $condition2)
}
```

is not equivalent to 
```
  name @include(if: $condition1) @include(if: $condition2)
```

We could use another "IR" there that can represent these cases. That felt a bit overkill just for this use case but maybe there can be other uses too.

Reply:
Yeah from GraphQL execution perspective, but from codegen we don't care right? we should treat this as one field as we going to generate only one `name` field.

Reply:
The thing is... apollo-android is also an execution engine, kind of. When reading from the cache, we're effectively acting like a server. If we don't honor directives, we will over fetch from the cache. Maybe it's not a big deal, maybe it is, I don't know...